### PR TITLE
use cache for slow query and statements

### PR DESCRIPTION
**Pull request by baurine**
- [x] use cache for slow query
- [x] use cache for statements

another solution for PR #822

### CODE REVIEW

File: ui/lib/apps/SlowQuery/utils/useSlowQueryTableController.ts [hunk 1]
```diff
@@ -137,37 +144,66 @@ export default function useSlowQueryTableController(
...
+        plans,
+        searchText,
+        timeRange,
+      } = queryOptions
+      const { desc, orderBy } = orderOptions
+      const { beginTime, endTime } = queryTimeRange
+      const cacheKey = `${schemas.join('_')}_${digest}_${limit}_${plans.join(
+        '_'
+      )}_${searchText}_${stringfyTimeRange(timeRange)}_${desc}_${orderBy}`
+      const cacheItem = slowQueryCacheMgr?.get(cacheKey)
```
Comment (breeswish) [inline ui/lib/apps/SlowQuery/utils/useSlowQueryTableController.ts:162]: What if user just clicks refresh (and there is new SQL statements in the time range)?

Comment (baurine) [reply ui/lib/apps/SlowQuery/utils/useSlowQueryTableController.ts:162]: The refresh will trigger force fetch:

```
  function refresh() {
    setErrors([])
    fetchSlowQueryList(true)
}
```

Comment (breeswish) [reply ui/lib/apps/SlowQuery/utils/useSlowQueryTableController.ts:162]: Cool

Comment (baurine) [reply ui/lib/apps/SlowQuery/utils/useSlowQueryTableController.ts]: Update the strategy by removing the cache when refreshing to simplify the code change and logic:

```js
  function refresh() {
    cacheMgr?.remove(cacheKey)

    setErrors([])
    setRefreshTimes((prev) => prev + 1)
}
```

**Comment by breeswish** [comment]
@unbyte PTAL, thanks

**Comment by unbyte** [comment]
LGTM

**Comment by breeswish** [comment]
Currently I'm going to leave this PR not picked into master, until a position memorization is implemented, since in production user is very likely to scroll, and not memorizing the position is worse than simply open a new window. @baurine PTAL

**Comment by baurine** [comment]
Got it!

**Pull request merged**